### ISSUE TITLE
[a11y]: File uploader file delete button needs more context

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#11198]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

n/a

### Automated testing tool and ruleset

inspect

### Description

As discussed at CAG, files uploaded through the File Uploader component can be removed, but the file's name is not part of the delete mechanism.
When multiple files are uploaded, all the delete mechanisms ('x') are just called "Delete" (or sometimes "clear').

Since there is no confirmation prompt before deleting, it is especially important that a user who cannot see has an ability to know which file is being deleted. 
![image](https://user-images.githubusercontent.com/14298245/162801237-b974cad0-6361-4b04-ab1d-d464bf583579.png)

![image](https://user-images.githubusercontent.com/14298245/162801973-ec1734ca-c757-4144-97f7-57162ffd11f3.png)

There are several ways of resolving, this, including constructing each uploaded file as an object. However, the simplest thing to do is:

1. Consistently change the default aria-label to "delete" instead of 'clear'.
2. provide the file name through aria-describedby

![image](https://user-images.githubusercontent.com/14298245/162802658-625108ee-ffcb-443a-8284-bdf3c6d7e1fa.png)

`<span class="cds--file__selected-file cds--file__selected-file--md"><p id="file1" class="cds--file-filename">button_usage_25.png</p><span class="cds--file__state-container"><button aria-label="Delete file" class="cds--file-close" aria-describedby="file1" type="button" tabindex="0"><svg focusable="false" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" fill="currentColor" width="16" height="16" viewBox="0 0 32 32" aria-hidden="true"><path d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"></path></svg></button></span></span>`

### WCAG 2.1 Violation

1.3.1 Info and relationships

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader

### Steps to reproduce

n/a

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
